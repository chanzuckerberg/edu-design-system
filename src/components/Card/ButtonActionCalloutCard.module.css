@import '../../../src/design-tokens/mixins.css';

/*------------------------------------*\
    # BUTTON ACTION CALLOUT CARD
\*------------------------------------*/

/**
 * Card elevated to drive attention and give an action.
 */

/**
 * Button action callout card body inner
 *
 * Wrapper around the description and button.
 */
.button-action-callout-card__body-inner {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  gap: var(--eds-size-2);
  margin-top: var(--eds-size-half);

  @media all and (min-width: $eds-bp-md) {
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }
}
