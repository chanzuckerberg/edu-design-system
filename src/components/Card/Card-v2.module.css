/*------------------------------------*\
    # CARD
\*------------------------------------*/

/**
 * A card is a block that typically contains a title, image, text, and/or calls to action.
 * The `:where` pseudo class function allows easy overriding via className.
 */
:where(.card) {
  position: relative;
  display: flex;
  flex-direction: column;

  padding: 1.5rem;
  height: 100%;
  border: calc(var(--eds-theme-border-width) * 1px) solid;


  &.card--container-style-low {
    border-color: var(--eds-theme-color-border-utility-default-low-emphasis);
  }

  &.card--container-style-medium {
    border-color: var(--eds-theme-color-border-utility-default-medium-emphasis)
  }

  &.card--container-style-high {
    border-color: var(--eds-theme-color-border-utility-default-high-emphasis);
  }

  &.card--container-style-none {
    border-color: transparent;
  }

  &.card--background-default {
    background-color: var(--eds-theme-color-background-utility-container);

    &:hover {
      background-color: var(--eds-theme-color-background-utility-container-hover);
    }

    &:active {
      background-color: var(--eds-theme-color-background-utility-container-active);
    }
  }

  &.card--background-call-out {
    background-color: var(--eds-theme-color-background-utility-information-low-emphasis);

    &:hover {
      background-color: var(--eds-theme-color-background-utility-information-low-emphasis-hover);
    }

    &:active {
      background-color: var(--eds-theme-color-background-utility-information-low-emphasis-active);
    }
  }

  &.card--background-call-out.card--container-style-medium {
    border-color: var(--eds-theme-color-border-utility-informational);
  }

  &:focus-visible {
    outline-offset: 1px;
    outline: 0.125rem solid var(--eds-theme-color-border-utility-focus);
  }

  /**
   * Invalid States
   
  &.card--container-style-low.card--background-call-out {
    border-color: fuchsia;
    border-style: dashed;
  }
  */
}

.card__header {
  display: flex;
  
  &.header--size-sm {
    gap: 0.5rem
  }

  &.header--size-md {
    gap: 1rem;
  }
}

.card__top-stripe {
  position: absolute;
  top: 0;
  left: 0;

  width: 100%;
}

:where(.card__top-stripe) {
  background-color: var(--eds-theme-color-background-utility-default-high-emphasis);
}

.top-stripe--medium {
  height: 0.5rem;
}

.top-stripe--high {
  height: 1rem;
}

.header__icon,
.header__action {
  flex-grow: 0;
}

.header__text {
  flex-grow: 2;
}

.card__body {
  flex-grow: 1;

  color: var(--eds-theme-color-text-utility-default-primary);
}

.header__eyebrow {
  color: var(--eds-theme-color-text-utility-default-secondary);
}

.header__title {
  color: var(--eds-theme-color-text-utility-default-primary);
}

.header__sub-title {
  color: var(--eds-theme-color-text-utility-default-secondary);
}