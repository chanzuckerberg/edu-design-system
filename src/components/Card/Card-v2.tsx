import clsx from 'clsx';
import type { HTMLAttributes, ReactNode } from 'react';
import React from 'react';

import type { Size } from '../../util/variant-types';
import type { IconName } from '../Icon';

import Icon from '../Icon';
import Text from '../Text';

import styles from './Card-v2.module.css';

export interface CardProps extends HTMLAttributes<HTMLElement> {
  // Component API
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  // Design API
  /**
   *
   * **Default is `"default"`**.
   */
  background: 'default' | 'call-out';
  /**
   * The bounding box and other container emphasis details
   *
   * **Default is `"low"`**.
   */
  containerStyle: 'none' | 'low' | 'medium' | 'high';
  /**
   * Decorative top bar used to cause a highlight on a given card. Use
   * utility classes to adjust background color.
   *
   * **Default is `"none"`**.
   */
  topStripe: 'none' | 'medium' | 'high';
  /**
   * Class to adjust top stripe color.
   */
  topStripeClassName: string;
}

export interface CardSubComponentProps {
  // Component API
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
}

export interface CardHeaderProps {
  // Component API
  /**
   * Child node(s) that can be nested inside component. Used in place of any of the above named slots.
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  // Design API
  /**
   * Component slot to add in an action-focused area to a card. Typically a button with hidden options.
   */
  action?: ReactNode;
  /**
   * Text above the main title of the card, to add category text/information.
   */
  eyebrow?: string;
  /**
   * Card slot for an icon sitting in front of the card header text
   */
  icon?: IconName;
  /**
   * Overall size treatment of the Card header
   */
  size?: Extract<Size, 'sm' | 'md'>;
  /**
   * Text below the main title of the card, to add supplementary information about the title
   */
  subTitle?: string;
  /**
   * Main title of the card
   */
  title?: string;
}

/**
 * `import {Card} from "@chanzuckerberg/eds";`
 *
 * Card component is the outer wrapper for the block that typically contains a title, image,
 * text, and/or calls to action.
 */
export const Card = ({
  background = 'default',
  className,
  children,
  containerStyle = 'low',
  topStripe = 'none',
  topStripeClassName,
  ...other
}: CardProps) => {
  const componentClassName = clsx(
    styles['card'],
    styles[`card--container-style-${containerStyle}`],
    styles[`card--background-${background}`],
    className,
  );
  return (
    <article className={componentClassName} {...other}>
      {children}
      {topStripe && (
        <div
          className={clsx(
            styles['card__top-stripe'],
            styles[`top-stripe--${topStripe}`],
            topStripeClassName,
          )}
        ></div>
      )}
    </article>
  );
};

/**
 * Body of the Card component.
 */
const CardBody = ({ children, className, ...other }: CardSubComponentProps) => {
  const componentClassName = clsx(styles['card__body'], className);
  return (
    <div className={componentClassName} {...other}>
      {children}
    </div>
  );
};

/**
 * Footer of the Card component.
 */
const CardFooter = ({
  children,
  className,
  ...other
}: CardSubComponentProps) => {
  const componentClassName = clsx(styles['card__footer'], className);
  return (
    <footer className={componentClassName} {...other}>
      {children}
    </footer>
  );
};

/**
 * Header of the Card component.
 */
const CardHeader = ({
  action,
  children,
  className,
  eyebrow,
  icon,
  size = 'md',
  subTitle,
  title,
  ...other
}: CardHeaderProps) => {
  const componentClassName = clsx(
    styles['card__header'],
    size && styles[`header--size-${size}`],
    className,
  );
  const headerEyebrowClassName = clsx(
    styles['header__eyebrow'],
    size && styles[`header--size-${size}`],
  );
  const headerTitleClassName = clsx(
    styles['header__title'],
    size && styles[`header--size-${size}`],
  );
  const headerSubTitleClassName = clsx(
    styles['header__sub-title'],
    size && styles[`header--size-${size}`],
  );

  return children ? (
    <header className={componentClassName} {...other}>
      {children}
    </header>
  ) : (
    <header className={componentClassName} {...other}>
      {icon && (
        <div className={styles['header__icon']}>
          <Icon
            name={icon}
            purpose="decorative"
            size={size === 'sm' ? '1rem' : '1.5rem'}
          />
        </div>
      )}
      <div className={styles['header__text']}>
        {eyebrow && (
          <Text
            as="div"
            className={headerEyebrowClassName}
            preset={size === 'sm' ? 'overline-sm' : 'overline-md'}
          >
            {eyebrow}
          </Text>
        )}
        {title && (
          <Text
            as="div"
            className={headerTitleClassName}
            preset={size === 'sm' ? 'title-sm' : 'headline-sm'}
          >
            {title}
          </Text>
        )}
        {subTitle && (
          <Text
            as="div"
            className={headerSubTitleClassName}
            preset={size === 'sm' ? 'body-sm' : 'body-lg'}
          >
            {subTitle}
          </Text>
        )}
      </div>
      {action && <div className={styles['header__action']}>{action}</div>}
    </header>
  );
};

Card.displayName = 'Card';
CardBody.displayName = 'Card.Body';
CardFooter.displayName = 'Card.Footer';
CardHeader.displayName = 'Card.Header';

Card.Body = CardBody;
Card.Footer = CardFooter;
Card.Header = CardHeader;
