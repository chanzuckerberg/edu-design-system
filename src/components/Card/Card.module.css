@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    #CARD
\*------------------------------------*/

/**
 * 1) A card is a block that typically contains a title, image,
 *    text, and/or calls to action
 */
.card {
  display: flex;
  flex-direction: column;
  padding: var(--eds-size-2);
  height: 100%;
  border-radius: var(--eds-border-radius-lg);
  border-width: var(--eds-theme-border-width);
  border-style: solid;
  border-color: var(--eds-theme-color-border-neutral-subtle);
  background-color: var(--eds-theme-color-background-neutral-default);
}

/**
 * Linked card
 * 1) Card that is wrapped in an a tag
 */
a.card {
  appearance: none;
  text-decoration: none;
  color: var(--eds-theme-color-text-neutral-default);
  transition: box-shadow var(--eds-anim-fade-quick) var(--eds-anim-ease);

  &:hover {
    box-shadow: var(--eds-theme-box-shadow);
  }
}

/**
 * Raised card
 * 1) Card that appears raised to show interactivity/emphasis
 */
.card--raised {
  box-shadow: var(--eds-box-shadow-md);
  border-color: var(--eds-theme-color-background-neutral-default);
}

/**
 * Card that is being dragged in drag and drop
 * 1) Passed down to card to be reused. 
 */
.card.eds-is-dragging {
  box-shadow: var(--eds-box-shadow-xl);
}

/**
 * Horizontal card
 * 1) Change the orientation to display header, body, and footer in a row
 */
.card--horizontal {
  flex-direction: row;
}

/**
 * Centered card
 * 1) Card where the content is centered
 */
.card--center {
  text-align: center;
}

/**
 * Card Inverted
 * 1) Used on dark backgrounds to lighten the text color and remove a box shadow
 */
.card--inverted {
  color: var(--eds-theme-color-text-neutral-default-inverse);
  background: transparent;
  border-color: var(--eds-theme-color-border-neutral-strong);
  box-shadow: none;
}

/**
 * Card header
 * 1) Add spacing in between the header and body
 */
.card__header {
  margin-bottom: var(--eds-size-2); /* 1 */

  /**
  * Card header within horizontal card
  * 1) Remove default spacing for vertical card. Add margin right for spacing between header and body
  */
  .card--horizontal & {
    margin-bottom: 0; /* 1 */
    margin-right: var(--eds-size-2); /* 1 */
  }
}

/**
 * Card body
 */
.card__body {
  flex: 1;
}

/**
 * Card footer
 * 1) Using margin-top since card__footer is
 *    optional and may not always exist. Putting
 *    margin-bottom on card__body would result in
 *    a gap if card__footer does not exist
 */
.card__footer {
  margin-top: var(--eds-size-2); /* 1 */

  /**
  * Card footer within horizontal card
  * 1) Using margin left to space away from card body. Zero out margin top.
  */
  .card--horizontal & {
    margin-top: 0; /* 1 */
    margin-left: var(--eds-size-2); /* 1 */
  }
}
