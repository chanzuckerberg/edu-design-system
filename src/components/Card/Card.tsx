import clsx from 'clsx';
import type { HTMLAttributes, ReactNode } from 'react';
import React from 'react';
import styles from './Card.module.css';

export interface CardProps extends HTMLAttributes<HTMLElement> {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Elevation variants
   * - **none** renders the card with no elevation (no box-shadow applied)
   * - **raised** renders the card that is raised off of the canvas (box-shadow applied)
   * - **dragging** renders the card that is raised even further off the canvas (during drag)
   *
   * **Default is `'none'`**.
   */
  elevation?: 'none' | 'raised' | 'dragging';
  /**
   * Property to apply a "dragging" elevation. Can be use while card is in a moving state.
   *
   * **Default is `false`**.
   */
  isDragging?: boolean;
}

export interface CardSubComponentProps {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
}

/**
 * `import {Card} from "@chanzuckerberg/eds";`
 *
 * Card component is the outer wrapper for the block that typically contains a title, image,
 * text, and/or calls to action.
 */
export const Card = ({
  className,
  children,
  elevation = 'none',
  isDragging = false,
  ...other
}: CardProps) => {
  const componentClassName = clsx(
    styles['card'],
    elevation === 'raised' && styles['card--raised'],
    elevation === 'dragging' && styles['card--dragging'],
    isDragging && styles['card--dragging'],
    className,
  );
  return (
    <article className={componentClassName} {...other}>
      {children}
    </article>
  );
};

/**
 * Body of the Card component.
 */
const CardBody = ({ children, className, ...other }: CardSubComponentProps) => {
  const componentClassName = clsx(styles['card__body'], className);
  return (
    <div className={componentClassName} {...other}>
      {children}
    </div>
  );
};

/**
 * Footer of the Card component.
 */
const CardFooter = ({
  children,
  className,
  ...other
}: CardSubComponentProps) => {
  const componentClassName = clsx(styles['card__footer'], className);
  return (
    <footer className={componentClassName} {...other}>
      {children}
    </footer>
  );
};

/**
 * Header of the Card component.
 */
const CardHeader = ({
  children,
  className,
  ...other
}: CardSubComponentProps) => {
  const componentClassName = clsx(styles['card__header'], className);
  return (
    <header className={componentClassName} {...other}>
      {children}
    </header>
  );
};

Card.displayName = 'Card';
CardBody.displayName = 'Card.Body';
CardFooter.displayName = 'Card.Footer';
CardHeader.displayName = 'Card.Header';

Card.Body = CardBody;
Card.Footer = CardFooter;
Card.Header = CardHeader;
