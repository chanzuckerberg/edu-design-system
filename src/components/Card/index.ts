export { Card as default } from './Card';
export { Card as CardV2 } from './Card';
