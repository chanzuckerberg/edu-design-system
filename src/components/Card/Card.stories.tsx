import { Meta } from '@storybook/react';
import React from 'react';
import { Card } from './Card';
import CardBody from '../CardBody';
import CardFooter from '../CardFooter';
import CardHeader from '../CardHeader';

export default {
  title: 'Molecules/Blocks/Card',
  component: Card,
} as Meta;

export const Default = () => (
  <Card>
    <CardHeader>
      <div className="fpo">Card Header</div>
    </CardHeader>
    <CardBody>
      <div className="fpo">Card Body</div>
    </CardBody>
    <CardFooter>
      <div className="fpo">Card Footer</div>
    </CardFooter>
  </Card>
);

export const Inverted = () => (
  <div style={{ backgroundColor: 'black', padding: '1rem' }}>
    <Card inverted={true}>
      <CardHeader>
        <div className="fpo">Card Header</div>
      </CardHeader>
      <CardBody>
        <div className="fpo">Card Body</div>
      </CardBody>
      <CardFooter>
        <div className="fpo">Card Footer</div>
      </CardFooter>
    </Card>
  </div>
);
