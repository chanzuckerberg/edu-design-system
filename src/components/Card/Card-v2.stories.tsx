import type { StoryObj, Meta } from '@storybook/react';
// import { userEvent, within } from '@storybook/testing-library';
import React from 'react';

import { Card } from './Card-v2';

import { ButtonV2 as Button } from '../Button';
import { ButtonGroupV2 as ButtonGroup } from '../ButtonGroup';
import Icon from '../Icon';
import { MenuV2 as Menu } from '../Menu';
import { Text } from '../Text/Text';

export default {
  title: 'Components/V2/Card',
  component: Card,
  parameters: {
    badges: ['intro-1.0', 'current-2.0'],
  },
  decorators: [(Story) => <div className="p-8">{Story()}</div>],
  args: {
    children: (
      <>
        <Card.Header>
          <div className="fpo w-full">Card Header</div>
        </Card.Header>
        <Card.Body>
          <div className="fpo">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <div className="fpo">Card Footer</div>
        </Card.Footer>
      </>
    ),
  },
  argTypes: {
    children: {
      control: {
        type: null,
      },
    },
    topStripeColor: {
      options: [
        '',
        'bg-brand-blue-1',
        'bg-brand-blue-2',
        'bg-brand-blue-3',
        'bg-brand-green',
        'bg-brand-orange',
        'bg-brand-pink',
        'bg-brand-purple',
        'bg-brand-red',
        'bg-brand-yellow',
      ],
      control: {
        type: 'select',
      },
    },
  },
} as Meta<Args>;

type Args = React.ComponentProps<typeof Card>;

/**
 * Cards come with structural containers for semantic grouping.
 */
export const Default: StoryObj<Args> = {};

export const WithFullHeader: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Card.Header
          eyebrow="Recommended for you"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <div className="fpo">Card Footer</div>
        </Card.Footer>
      </>
    ),
  },
};

export const WithFullHeaderAndIcon: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Card.Header
          eyebrow="Recommended for you"
          icon="person-encircled"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <div className="fpo">Card Footer</div>
        </Card.Footer>
      </>
    ),
  },
};

function CardMenu() {
  return (
    <Menu>
      <Menu.PlainButton>
        <Icon
          name="dots-horizontal"
          purpose="informative"
          size="2rem"
          title="show more"
        />
      </Menu.PlainButton>
      <Menu.Items data-testid="menu-content">
        <Menu.Item
          href="https://headlessui.com/react/menu#menu-button"
          icon="link"
        >
          Headless UI Docs
        </Menu.Item>
        <Menu.Item
          href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu"
          icon="link"
        >
          MDN: Menu
        </Menu.Item>
        <Menu.Item href="#index" onClick={() => console.log('Item clicked')}>
          Trigger Action
        </Menu.Item>
        <Menu.Item disabled href="https://example.org/" icon="warning-filled">
          Not Possible (disabled)
        </Menu.Item>
      </Menu.Items>
    </Menu>
  );
}

export const WithSmallFullHeaderAndIcon: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Card.Header
          action={<CardMenu />}
          eyebrow="Recommended for you"
          icon="person-encircled"
          size="sm"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <div className="fpo">Card Footer</div>
        </Card.Footer>
      </>
    ),
  },
};

export const WithHorizontalPrimaryButton: StoryObj<Args> = {
  args: {
    children: (
      <>
        <Card.Header
          action={<CardMenu />}
          eyebrow="Recommended for you"
          size="md"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo my-4">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <ButtonGroup>
            {/* This has to be manually tested since Tooltip tests are flaky in Chromatic */}
            <Button onClick={() => {}} rank="primary">
              Primary Action
            </Button>
          </ButtonGroup>
        </Card.Footer>
      </>
    ),
  },
};

export const CustomBrandCard: StoryObj<Args> = {
  args: {
    containerColor: 'custom-brand',
    className: 'border-brand-red bg-brand-red',
  },
};

export const TopStripe: StoryObj<Args> = {
  args: {
    topStripe: 'medium',
    children: (
      <>
        <Card.Header
          action={<CardMenu />}
          eyebrow="Recommended for you"
          size="md"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo my-4">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <ButtonGroup>
            {/* This has to be manually tested since Tooltip tests are flaky in Chromatic */}
            <Button onClick={() => {}} rank="primary">
              Primary Action
            </Button>
          </ButtonGroup>
        </Card.Footer>
      </>
    ),
  },
};

export const CustomTopStripe: StoryObj<Args> = {
  args: {
    topStripe: 'high',
    topStripeColor: 'bg-brand-purple',
    children: (
      <>
        <Card.Header
          action={<CardMenu />}
          eyebrow="Recommended for you"
          size="md"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo my-4">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <ButtonGroup>
            {/* This has to be manually tested since Tooltip tests are flaky in Chromatic */}
            <Button onClick={() => {}} rank="primary">
              Primary Action
            </Button>
          </ButtonGroup>
        </Card.Footer>
      </>
    ),
  },
};

export const BackgroundCallout: StoryObj<Args> = {
  args: {
    containerColor: 'call-out',
    children: (
      <>
        <Card.Header
          action={<CardMenu />}
          eyebrow="Recommended for you"
          size="md"
          subTitle="Get to know your colleagues"
          title="Question of the day"
        />
        <Card.Body>
          <div className="fpo my-4">Card Body</div>
        </Card.Body>
        <Card.Footer>
          <ButtonGroup>
            {/* This has to be manually tested since Tooltip tests are flaky in Chromatic */}
            <Button onClick={() => {}} rank="primary">
              Primary Action
            </Button>
          </ButtonGroup>
        </Card.Footer>
      </>
    ),
  },
};

/**
 * Cards have hierarchy, and can contain other cards with one level. When using a nested card, you can use `.ChildCard`,
 * which comes preconfigured with some settings and defaults applied. All the sub-components of card work within `.Childcard`.
 */
export const ChildCards: StoryObj<Args> = {
  args: {
    containerColor: 'call-out',
    topStripe: 'high',
    containerStyle: 'high',
    children: (
      <>
        <Card.Header size="md" title="Card Group"></Card.Header>
        <Card.Body>
          <Card.ChildCard draggable="true">
            <Card.Header
              action={<CardMenu />}
              eyebrow="Eyebrow Text"
              subTitle="Sub-title text"
              title="Title text"
            />
          </Card.ChildCard>
        </Card.Body>
      </>
    ),
  },
};

/**
 * Cards can be marked as currently being dragged. This can be used in combination with the `draggable` attribute from HTML,
 * or to work with a drag-n-drop library.
 */
export const WhileDragging: StoryObj<Args> = {
  args: {
    isDragging: true,
  },
};

/**
 * Implementation Example: Cancelling a card membership
 */
export const CancelMembership: StoryObj<Args> = {
  parameters: {
    badges: ['intro-1.0', 'current-2.0', 'implementationExample'],
  },
  args: {
    children: (
      <>
        <Card.Header
          size="sm"
          subTitle="We're sad to see you go"
          title="Cancel membership?"
        />
        <Card.Body>
          <Text className="py-4" preset="body-sm">
            Lorem ipsum dolor sit amet consectetur. Id pretium consequat
            consequat aliquam arcu
          </Text>
        </Card.Body>
        <Card.Footer>
          <ButtonGroup>
            {/* This has to be manually tested since Tooltip tests are flaky in Chromatic */}
            <Button
              onClick={() => {}}
              rank="primary"
              size="md"
              variant="critical"
            >
              Confirm cancellation
            </Button>
          </ButtonGroup>
        </Card.Footer>
      </>
    ),
  },
};
