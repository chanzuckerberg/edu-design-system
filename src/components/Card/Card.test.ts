import { generateSnapshots } from '@chanzuckerberg/story-utils';
import type { StoryFile } from '@storybook/testing-react';
import * as stories from './Card.stories';

describe('<Card />', () => {
  generateSnapshots(stories as StoryFile);
});
