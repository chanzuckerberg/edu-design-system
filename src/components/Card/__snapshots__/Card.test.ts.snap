// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Default story renders snapshot 1`] = `
<article
  class="card"
  data-bootstrap-override="card"
>
  <header
    class="card__header"
  >
    <div
      class="fpo"
    >
      Card Header
    </div>
  </header>
  <div
    class="card__body"
  >
    <div
      class="fpo"
    >
      Card Body
    </div>
  </div>
  <footer
    class="card__footer"
  >
    <div
      class="fpo"
    >
      Card Footer
    </div>
  </footer>
</article>
`;

exports[`<Card /> Horizontal story renders snapshot 1`] = `
<article
  class="card card--horizontal"
  data-bootstrap-override="card"
>
  <header
    class="card__header"
  >
    <div
      class="fpo"
    >
      Card Header
    </div>
  </header>
  <div
    class="card__body"
  >
    <div
      class="fpo"
    >
      Card Body
    </div>
  </div>
  <footer
    class="card__footer"
  >
    <div
      class="fpo"
    >
      Card Footer
    </div>
  </footer>
</article>
`;

exports[`<Card /> Raised story renders snapshot 1`] = `
<article
  class="card card--raised"
  data-bootstrap-override="card"
>
  <header
    class="card__header"
  >
    <div
      class="fpo"
    >
      Card Header
    </div>
  </header>
  <div
    class="card__body"
  >
    <div
      class="fpo"
    >
      Card Body
    </div>
  </div>
  <footer
    class="card__footer"
  >
    <div
      class="fpo"
    >
      Card Footer
    </div>
  </footer>
</article>
`;
